tests/OAuth2Test.php

```diff
@@ -1078,3 +1080,70 @@ public function testShouldReturnAValidIdToken()
         $this->assertEquals($origIdToken['aud'], $roundTrip->aud);
     }
 }
+
+class OAuth2StsTest extends TestCase
+{
+    use ProphecyTrait;
+
+    private $publicKey;
+    private $privateKey;
+    private $stsMinimal = [
+        'tokenCredentialUri' => 'https://tokens_r_us/test',
+        'subjectTokenType' => 'urn:ietf:params:aws:token-type:aws4_request',
+    ];
+
+    public function testStsGrantType()
+    {
+        $credentialSource = $this->prophesize(CredentialSourceInterface::class);
+        $o = new OAuth2($this->stsMinimal + ['subjectTokenFetcher' => $credentialSource->reveal()]);
+        $this->assertEquals(OAuth2::STS_URN, $o->getGrantType());
+    }
+
+    public function testStsCredentialsRequestMinimal()
+    {
+        $credentialSource = $this->prophesize(CredentialSourceInterface::class);
+        $credentialSource->fetchSubjectToken(null)
+            ->shouldBeCalledOnce()
+            ->willReturn('xyz');
+        $o = new OAuth2($this->stsMinimal + ['subjectTokenFetcher' => $credentialSource->reveal()]);
+        $request = $o->generateCredentialsRequest();
+        $this->assertEquals('POST', $request->getMethod());
+        $this->assertEquals($this->stsMinimal['tokenCredentialUri'], (string) $request->getUri());
+        parse_str((string)$request->getBody(), $requestParams);
+        $this->assertCount(4, $requestParams);
+        $this->assertEquals(OAuth2::STS_URN, $requestParams['grant_type']);
+        $this->assertEquals('xyz', $requestParams['subject_token']);
+        $this->assertEquals($this->stsMinimal['subjectTokenType'], $requestParams['subject_token_type']);
+    }
+
+    public function testStsCredentialsRequestFull()
+    {
+        $credentialSource = $this->prophesize(CredentialSourceInterface::class);
+        $credentialSource->fetchSubjectToken(null)
+            ->shouldBeCalledOnce()
+            ->willReturn('xyz');
+        $stsMinimal = $this->stsMinimal + [
+            'subjectTokenFetcher' => $credentialSource->reveal(),
```

Comment:
Can you also add a test (probably in the ExternalAccountCredentialsTests) thais oauth2/sts object is getting the correct values from the credential configuration when it is instantiated?